cypress/e2e/1-testing/login-fail.cy.js

```diff
@@ -0,0 +1,19 @@
+describe("Login Fail", () => {
+  it("should show error message when logging in with invalid credentials", () => {
+    cy.visit("https://murots.github.io/Workflow_ca/");
```

Comment:
```suggestion
    cy.visit("/");
```

When you test a remote deployment, instead of the local code - you are not providing any utility at all. Tests should be run on "hot" code in your branch, rather than "cold" code that has already been deployed.

This is because the decision "should I deploy this" should only be "yes" if the work has been tested beforehand, not afterwards.

Cypress runs its own webserver, do you can just hit "/" instead.

Reply:
Ah, I must have missed that. Thank you. All passes now.